src/Domain/Direction.hpp

```diff
@@ -43,18 +43,21 @@ class Direction {
   double sign() const noexcept { return (Side::Lower == side_ ? -1.0 : 1.0); }
 
   /// The opposite Direction.
-  Direction<VolumeDim> opposite() const;
+  Direction<VolumeDim> opposite() const noexcept;
+
+  // An array of Directions.
+  static std::array<Direction<VolumeDim>, 2 * VolumeDim> directions() noexcept;
```

Comment:
Perhaps name this something like `all_directions` to make what it does more obvious.